core/src/mindustry/input/InputHandler.java

```diff
@@ -2032,7 +2032,7 @@ public void remove(){
 
     public void add(){
         Core.input.getInputProcessors().remove(i -> i instanceof InputHandler || (i instanceof GestureDetector && ((GestureDetector)i).getListener() instanceof InputHandler));
-        Core.input.addProcessor(detector = new GestureDetector(20, 0.5f, 0.3f, 0.15f, this));
+        Core.input.addProcessor(detector = new GestureDetector(20, 0.5f, (float) Core.settings.getInt("longpressduration") / 1000, 0.15f, this));
```

Comment:
This needs to be dynamically updated when the setting changes. In its current state, it requires a game restart, which is not acceptable.